aws/resource_aws_default_subnet.go

```diff
@@ -38,6 +38,7 @@ func resourceAwsDefaultSubnet() *schema.Resource {
 	// map_public_ip_on_launch is a computed value for Default Subnets
 	dsubnet.Schema["map_public_ip_on_launch"] = &schema.Schema{
 		Type:     schema.TypeBool,
+		Optional: true,
```

Comment:
For backwards compatibility I don't set the` Default: false` flag here as is done for the equivalent attribute in the `aws_subnet` resource.